File: 백준/Silver/24444. 알고리즘 수업 － 너비 우선 탐색 1/알고리즘 수업 － 너비 우선 탐색 1.py
import sys
from collections import deque
input = sys.stdin.readline
def bfs(x):
  global cnt
  q = deque()
  q.append(x)
  check[x] = cnt
  while q:
    x = q.popleft()
    a[x].sort()
    for i in a[x]:
      if not check[i]:
        cnt += 1
        check[i] = cnt
        q.append(i)
n,m,r = map(int,input().split())
a = [[] for _ in range(n+1)]
for _ in range(m):
  u,v = map(int,input().split())
  a[u].append(v)
  a[v].append(u)
check = [0]*(n+1)
cnt = 1
bfs(r)
for i in range(1,n+1):
  print(check[i])

File: 백준/Silver/24444. 알고리즘 수업 － 너비 우선 탐색 1/알고리즘 수업 － 너비 우선 탐색 1.cc
#include <iostream>
#include <algorithm>
#include <vector>
#include <queue>
using namespace std;
int n, m, st, cnt = 1;
vector<vector<int>> a;
vector<int>check;

void bfs(int x) {
	queue<int> q;
	q.push(x);
	check[x] = cnt;
	while (!q.empty()) {
		int x = q.front();
		q.pop();
		sort(a[x].begin(), a[x].end());
		for (int i : a[x]) {
			if (check[i] == 0) {
				cnt++;
				check[i] = cnt;
				q.push(i);
			}
		}
	}
}
int main() {
    cin >> n >> m >> st;
	a.resize(n + 1);
	check.resize(n + 1);
	for (int i = 0; i < m; i++) {
		int u, v;
		cin >> u >> v;
		a[u].push_back(v);
		a[v].push_back(u);
	}
	bfs(st);
	for (int i = 1; i <= n; i++) {
		cout << check[i] << '\n';
	}
    return 0;
}